sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3701,6 +3704,20 @@ def _check_existing_cluster(
                       if resources.cloud is not None else clouds.Cloud)
         task_cloud.check_cluster_name_is_valid(cluster_name)
 
+        if to_provision is None:
+            logger.info(
+                f'The cluster {cluster_name!r} was autodowned or manually '
+                'terminated on the cloud console. Using the original resources '
+                'to provision a new cluster.')
+            # The cluster is recently terminated either by autostop or manually
+            # terminated on the cloud. We should use the original resources to
+            # provision the cluster.
+            assert isinstance(previous_handle,
+                              CloudVmRayResourceHandle), (previous_handle,
+                                                          cluster_name)
```

Comment:
Style nit: The code looks a bit ugly tbh.
```suggestion
            is_cloud_vm_handle = isinstance(previous_handle,
                                                                      CloudVmRayResourceHandle)
            assert is_cloud_vm_handle, (previous_handle, cluster_name)
```

Reply:
Unfortunately, mypy is not clear enough to have this two-line assertion. I changed the output to be a string, so as to make it look a bit better. PTAL. : )